From 5989c8e961e1ad93b26da3616bd1aa907fb763a4 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Mon, 16 Feb 2026 18:58:59 +0000
Subject: [PATCH] Add projector support to agold-cmd and include in GSI

Update agold-cmd to support ioctl and cmd subcommands for controlling
the Unihertz/Agold projector hardware. The cmd() AIDL method takes an
IAgoldDaemon binder (not a callback), so pass the service reference.

Include agold-cmd in PRODUCT_PACKAGES so it ships in the GSI image.
---
 base.mk            |  3 ++-
 cmds/agold-cmd.cpp | 18 +++++++-----------
 2 files changed, 9 insertions(+), 12 deletions(-)

diff --git a/base.mk b/base.mk
index 1928cdc..2f9a77a 100644
--- a/base.mk
+++ b/base.mk
@@ -147,7 +147,8 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     lightsctl \
     lightsctl-aidl \
-    uevent
+    uevent \
+    agold-cmd
 
 PRODUCT_COPY_FILES += \
     device/phh/treble/files/adbd.rc:system/etc/init/adbd.rc
diff --git a/cmds/agold-cmd.cpp b/cmds/agold-cmd.cpp
index 9943557..a7e7ced 100644
--- a/cmds/agold-cmd.cpp
+++ b/cmds/agold-cmd.cpp
@@ -7,16 +7,12 @@ using ::android::sp;
 
 int main(int argc, char **argv) {
 	auto svc = android::checkVintfService<IAgoldDaemon>();
-    android::String16 a;
-    svc->getNotGsi(&a);
-    std::cout << "notgsi is " << a << std::endl;
-    if (argc >= 2) {
-        android::String16 b(argv[1]);
-        svc->setNotGsi(b);
-    }
-    svc->getNotGsi(&a);
-    std::cout << "notgsi is " << a << std::endl;
-
-    svc->setCameraClientPackageName(android::String16("com.mediatek.camera"));
+    int32_t ret = 0;
 
+    if (std::string(argv[1]) == "ioctl") {
+        svc->SendMessageToIoctl(atoi(argv[2]), atoi(argv[3]), atoi(argv[4]), atoi(argv[5]), &ret);
+    } else {
+        svc->cmd(android::String16(argv[2]), svc);
+    }
+    printf("Returned %d\n", ret);
 }
-- 
2.52.0

